internal/provider/resource_application_test.go

```diff
@@ -137,6 +137,41 @@ func TestAcc_ResourceApplication_Updates(t *testing.T) {
 	})
 }
 
+// TestAcc_ResourceApplication_UpdatesRevisionConfig will test the revision update that have new config parameters on
+// the charm. The test will check that the config is updated and the revision is updated as well.
+func TestAcc_ResourceApplication_UpdatesRevisionConfig(t *testing.T) {
+	modelName := acctest.RandomWithPrefix("tf-test-application")
+	appName := "github-runner"
+	resource.Test(t, resource.TestCase{
+		PreCheck:                 func() { testAccPreCheck(t) },
+		ProtoV6ProviderFactories: frameworkProviderFactories,
+		Steps: []resource.TestStep{
+			{
+				SkipFunc: func() (bool, error) {
+					return testingCloud != LXDCloudTesting, nil
+				},
+				Config: testAccResourceApplicationWithRevisionAndConfig(modelName, appName, 8, ""),
```

Comment:
Per the latest example: https://pastebin.ubuntu.com/p/ncp6NBpjT9/ let's start with revision 88 and upgrade to 95 in the test. The new charm config is `runner-storage`.  The charm revisions are significantly newer.